src/cfg/Relooper.cpp

```diff
@@ -107,12 +107,18 @@ static wasm::Expression* HandleFollowupMultiples(wasm::Expression* Ret,
 
 // Branch
 
-Branch::Branch(wasm::Expression* ConditionInit, wasm::Expression* CodeInit)
-  : Condition(ConditionInit), Code(CodeInit) {}
+Branch::Branch(Relooper* relooper,
+               wasm::Expression* ConditionInit,
+               wasm::Expression* CodeInit)
+  : Condition(ConditionInit), Code(CodeInit) {
+  relooper->Branches.push_back(std::unique_ptr<Branch>(this));
```

Comment:
I'm not sure it's a great idea to construct a `unique_ptr` from the `this` pointer. If a `Branch` is ever stack-allocated, then when it goes out of scope the `unique_ptr` will try to free the stack memory, which would be bad. It's also less efficient than calling `std::make_unique`.

The simple alternative would be to have all the callsites where a `Branch` is allocated be responsible for calling `std::make_unique`, adding the `unique_ptr` to the Relooper's vector, then extracting and using the raw pointer. But that's a lot to ask of multiple callers. Perhaps we can get the best of both options by adding utility methods to `Relooper` to do the allocations and handle all that work for the caller?

Reply:
Sure, sounds good. Going to give it a try :)

Reply:
Moved all of this into methods now, mirroring the respective constructor signatures: `Relooper::addBlock/addBranch/addXYShape`